### DISCUSSION TITLE
RobotClimber.java Integration

### DISCUSSION THREAD

**Discussion by patrick-philbin** [frc5567/2023-Robot#112] [Team Posts]
RobotClimber.java has been added to masterDev and has a pull request to be merged to master. The class is completely independent of Robot.java, meaning buttons will be called from Robot.java to trigger different methods.
Currently, the buttons to trigger methods will be:
Copilot Left Trigger - driveSpeed for the winchControl method.

Copilot Left Bumper - halfSpeed for the winchControl method (when held/toggled, the winch will move at half it's input speed).

Copilot DPad Up - Call the ExtendSolenoid method in order to extend the Solenoid.

Copilot DPad Down - Call the RetractSolenoid method in order to retract the Solenoid.

These buttons have not been implemented to Robot.java, and will need to be in the future.

**Comment by CooperTrout** [comment]
the DPad is not part of the Xbox Controller Class so instead we have decided to use the left joystick on the copilot  controller to extend and lower the solenoid.
